### ISSUE TITLE
Animebytes freeleech option

### ISSUE THREAD

**Issue by Entixs** [Prowlarr/Prowlarr#352]
**Is your feature request related to a problem? Please describe.**
No, it is not.

**Describe the solution you'd like**
I would like to see an option for Animebytes Indexer to search for freeleech only or an option to search for freeleech torrents first and if it doesn't find a freeleech to go with a non-freeleech.

**Comment by Qstick** [comment]
Does Animebytes support freeleech search on the site? If so what is the URL parameter used?

**Comment by ADRFranklin** [comment]
Yeah, it does. here is an example of the url with `freeleech=1` on the end of the url
`https://animebytes.tv/torrents.php?action=advanced&search_type=title&sort=relevance&way=desc&hentai=0&showhidden=0&freeleech=1`
@Qstick

**Comment by johnpyp** [comment]
@ADRFranklin I thought this was how it worked too, and the advanced search in Animebytes supports this, but it seems like filtering by freeleech doesn't actually exclude or prune non-freeleech results as far as I can tell. I still get results that aren't tagged with `| Freeleech`, and still have a `RawDownMultiplier` of 1.0.

One solution is filtering on our side by `RawDownMultiplier`, but that has the possibly undesirable side effect of emitting too few results when paginating, though it seems possible the Prowlarr pagination system might be able to handle that without too many problems.

**Comment by ADRFranklin** [comment]
I'm unsure of how prowlarr does it's scrapping, but if it's like any other scraper I have written in the past, could you not just look for results that contain `alt="Freeleech!"` inside the `<a>` tag ?

or actually there is a freeleech class attacthed to the `<tr>` element
```
<tr id="torrent_909048" class="torrent  group_torrents_5419 freeleech">
                        <td class="torrent_properties">
                            <span class="download_link">[<a href="https://animebytes.tv/torrent/909048/download/[hidden_user_id]" title="Download torrent">DL</a>]</span> &nbsp;
                            <a href="torrents.php?id=5419&amp;torrentid=909048">Web | MKV | h264 | 1080p | AAC 2.0 | Softsubs (KAN3D2M) | <img src="/static/common/flicon.png" alt="Freeleech!" title="This torrent is freeleech. Remember to seed!"></a>
                        </td>
                        <td class="torrent_size"><span>87.41 GiB</span></td>
                        <td class="torrent_snatched" title="Snatched">
                            <span>21</span></td>
                        <td class="torrent_seeders" title="Seeders">
                            <span>3</span></td>
                        <td class="torrent_leechers" title="Leechers">
                            <span>0</span></td>
                    </tr>
```

So we could search by the `freeleech` query param to make sure we only get results that have freeleech, and then we filter that content by elements that contain the `freeleech` class. This makes the most sense to me, what are your opinions?

**Comment by ADRFranklin** [comment]
@johnpyp What are your thoughts on this?

**Comment by johnpyp** [comment]
@ADRFranklin Yeah, seems like a good idea. We don't have to actually do that kind of parsing - that data is provided in the api for us (as the RawDownMultiplier being 0, or | Freeleech being in the title).

**Comment by Hayajiro** [comment]
I implemented this feature a couple days ago, as well as a fix to the way RSS is implemented for AB (only show recent uploads, not all of them). It's currently running in my private setup, but when everything continues to work as expected, I'll push those changes later and open a new PR.

Still need to get used to the Prowlarr codebase for one improvement, though:  
Right now, the RSS fix requires a second indexer configured just for RSS, as I haven't found a way to differentiate between RSS and regular searches in the request handling. However, I have to admit I haven't been looking thoroughly yet.

**Comment by fr3fou** [comment]
Have there been any updates to this?
